### ISSUE TITLE
Waiting on Thread interrupted waiting for connection acknowledgement

### ISSUE THREAD

**Issue by AliXhah** [aws-amplify/amplify-android#2511]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Not applicable

### Amplify Categories

GraphQL API

### Gradle script dependencies

<details>

```groovy
    amplify_api: ^0.6.0
    amplify_auth_cognito: ^0.6.0
    amplify_authenticator: ^0.2.2
    amplify_datastore: ^0.6.13
    amplify_flutter: ^0.6.0
    amplify_storage_s3: ^0.6.0

```

</details>

### Environment information

<details>

```
# Put output below this line
com.android.tools.build:gradle:4.1.0

```

</details>

### Please include any relevant guides or documentation you're referencing

Im working on flutter with AWS amplify

### Describe the bug

When app starts it prints 

I/flutter (11786): Amplify configuration Successfully.
I/amplify:aws-datastore(11786): Creating table: LastSyncMetadata
I/amplify:aws-datastore(11786): Creating table: PersistentModelVersion
I/amplify:aws-datastore(11786): Creating table: Accounts
I/amplify:aws-datastore(11786): Creating table: PersistentRecord
I/amplify:aws-datastore(11786): Creating table: AWSCommittee
I/amplify:aws-datastore(11786): Creating table: AWSPerson
I/amplify:aws-datastore(11786): Creating table: ModelMetadata
I/amplify:aws-datastore(11786): Creating index for table: PersistentRecord
I/amplify:flutter:datastore(11786): Unhandled DataStoreHubEvent: SUCCEEDED

I/amplify:aws-datastore(11786): Starting API synchronization mode.
I/amplify:aws-datastore(11786): Starting processing subscription events.
I/amplify:aws-datastore(11786): Orchestrator lock acquired.
I/amplify:aws-datastore(11786): DataStore plugin initialized.
I/amplify:aws-datastore(11786): Orchestrator lock released.

 Long monitor contention with owner pool-23-thread-1 (12480) at void com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(com.amplifyframework.api.graphql.GraphQLRequest, com.amplifyframework.api.aws.AuthorizationType, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Action)(SubscriptionEndpoint.java:188) waiters=0 in void com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(com.amplifyframework.api.graphql.GraphQLRequest, com.amplifyframework.api.aws.AuthorizationType, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Action) for 8.490s
E/amplify:aws-datastore(11786): Failure encountered while attempting to start API sync.
E/amplify:aws-datastore(11786): DataStoreException{message=DataStore subscriptionProcessor failed to start., cause=GraphQLResponseException{message=Subscription error for AWSPerson: [GraphQLResponse.Error{message='Validation error of type FieldUndefined: Field '_deleted' in type 'AWSPerson' is undefined @ 'onUpdateAWSPerson/_deleted'', locations='null', path='null', extensions='null'}, GraphQLResponse.Error{message='Validation error of type FieldUndefined: Field '_lastChangedAt' in type 'AWSPerson' is undefined @ 'onUpdateAWSPerson/_lastChangedAt'', locations='null', path='null', extensions='null'}, GraphQLResponse.Error{message='Validation error of type FieldUndefined: Field '_version' in type 'AWSPerson' is undefined @ 'onUpdateAWSPerson/_version'', locations='null', path='null', extensions='null'}], errors=[GraphQLResponse.Error{message='Validation error of type FieldUndefined: Field '_deleted' in type 'AWSPerson' is undefined @ 'onUpdateAWSPerson/_deleted'', locations='null', path='null', extensions='null'}, GraphQLResponse.Error{message='Validation error of type FieldUndefined: Field '_lastChangedAt' in type 'AWSPerson' is undefined @ 'onUpdateAWSPerson/_lastChangedAt'', locations='null', path='null', extensions='null'}, GraphQLResponse.Error{message='Validation error of type FieldUndefined: Field '_version' in type 'AWSPerson' is undefined @ 'onUpdateAWSPerson/_version'', locations='null', path='null', extensions='null'}], recoverySuggestion=See attached list of GraphQLResponse.Error objects.}, recoverySuggestion=Check your internet.}

Then again and again it prints 

Long monitor contention with owner pool-23-thread-1 (12480) at void com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(com.amplifyframework.api.graphql.GraphQLRequest, com.amplifyframework.api.aws.AuthorizationType, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Action)(SubscriptionEndpoint.java:188) waiters=1 in void com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(com.amplifyframework.api.graphql.GraphQLRequest, com.amplifyframework.api.aws.AuthorizationType, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Consumer, com.amplifyframework.core.Action) for 8.497s

### Code Snippet

```java
// Put your code below this line.

Future<void> login(context) async {
    try {
      Amplify.DataStore.observeQuery(Accounts.classType).listen((event) async {
        if (event.isSynced) {
          //boolean value
          turnOffAnimator();

          List<Accounts> awsAccounts = await Amplify.DataStore.query(
              Accounts.classType,
              where: Accounts.PHONENUMBER.eq(userNumber.trim()));

          if (awsAccounts.isNotEmpty) {
            Accounts awsAccount = awsAccounts.first;
            if (awsAccount.password.trim() == userPin.trim()) {
              turnOffAnimator();
              final prefs = await SharedPreferences.getInstance();
              await prefs.setString('userName', awsAccount.accountTitle.trim());
              await prefs.setString('phoneNumber', awsAccount.phoneNumber);

              turnOffAnimator();
              navigateToHome();
            } else {
              _snackbarService.showSnackbar(
                  message:
                      'User number or password is not correct, Please try again later.');
            }
            turnOffAnimator();
          } else {
            turnOffAnimator();
            _snackbarService.showSnackbar(
                message:
                    'User number or password is not correct, Please try again later.');
          }
        } else {
          turnOnAnimator();
        }
      });
    } catch (e) {
      print('Signin error: $e');
      turnOffAnimator();
      _snackbarService.showSnackbar(
          message:
              'User number or password is not correct, Please try again later.');
      return false;
    }
  }
}
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by mattcreaser** [comment]
@AliXhah the `_deleted`, `_lastChangedAt`, and `_version` fields are used in Datastore's conflict detection feature. This must be enabled to use Datastore. You can double check if conflict detection is enabled in the AppSync console by inspecting your schema to see if those fields are present, or by looking at `Data sources > AWSPersonTable` and seeing if `Enable data source versioning` is enabled.

I would suggest running `amplify update api` and selecting `GraphQL` and then `Enable conflict detection (required for DataStore)`. Be sure to push the changes.

> Long monitor contention with owner pool-23-thread-1 (12480) at void com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription

This is an expected log message and not cause for concern. We have a synchronization lock that is held for some time while establishing subscriptions.

**Comment by mattcreaser** [comment]
Closing due to inactivity. Please feel free to open another issue if required, thanks!
